src/gallium/state_trackers/nine/texture9.c

```diff
@@ -309,6 +309,13 @@ NineTexture9_AddDirtyRect( struct NineTexture9 *This,
         pDirtyRect ? pDirtyRect->left : 0, pDirtyRect ? pDirtyRect->top : 0,
         pDirtyRect ? pDirtyRect->right : 0, pDirtyRect ? pDirtyRect->bottom : 0);
 
+    if (pDirtyRect && (pDirtyRect->left < 0 ||
```

Comment:
For LockRect, the behaviour is different on XP and Vista+.
I guess it is the same for dirtyRect.
Do you have an example of game that requires the behaviour introduced by this commit ?